pilot/cmd/pilot-discovery/main.go

```diff
@@ -213,22 +240,26 @@ var (
 				return fmt.Errorf("failed to create discovery service: %v", err)
 			}
 
-			// Set up configuration validation admission
-			// controller. Fill in remaining admission controller
-			// options
-			flags.admissionArgs.Descriptor = configClient.ConfigDescriptor()
-			flags.admissionArgs.ServiceNamespace = flags.namespace
-			flags.admissionArgs.DomainSuffix = flags.controllerOptions.DomainSuffix
-			flags.admissionArgs.ValidateNamespaces = []string{
-				flags.controllerOptions.WatchedNamespace,
-				flags.namespace,
-			}
-			admissionController, err := admit.NewController(client, flags.admissionArgs)
-			if err != nil {
-				return fmt.Errorf("failed to create validation admission controller: %v", err)
+			if flags.configStore == kubernetesConfigStore && registered[platform.KubernetesRegistry] {
```

Comment:
AdmissionController creation should only be predicated on the configStore. It's possible to use k8s ConfigStore with Consul/Eureka service registry in which case we'd still want to create the admission controller.

Reply:
@ayj Updated as per conversation [here](https://github.com/istio/old_pilot_repo/issues/1566#issuecomment-340972929).